beacon-chain/core/validators/validator.go

```diff
@@ -497,12 +499,15 @@ func ProcessDeposit(
 	// TODO(#258): Validate proof of possession using BLS.
 	var publicKeyExists bool
 	var existingValidatorIndex int
-	for idx, val := range state.ValidatorRegistry {
-		if bytes.Equal(val.GetPubkey(), pubkey) {
-			publicKeyExists = true
-			existingValidatorIndex = idx
-		}
-	}
+
+	existingValidatorIndex, publicKeyExists = validatorIndexMap[stateutils.BytesToBytes32(pubkey)]
+
+	//for idx, val := range state.ValidatorRegistry {
```

Comment:
Why is this commented out?

Reply:
Oh, because it is accessed in O(1) time on line 503. I'll delete this comment block. Thanks!